src/main/java/com/google/sps/data/User.java

```diff
@@ -42,8 +42,9 @@ else if(input == "mentor" || input == "MENTOR") {
       return result;
   }
 
-  public boolean equals(User user) {
-    return this.name == user.name;
+  public boolean equals(Object user) {
```

Comment:
Because of this: This method returns true if this collection contains the specified element. More formally, returns true if and only if this collection contains at least one element e such that (o==null ? e==null : o.equals(e)). It compares the object to the user List when calling contains() for the comparisons in the tests